src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/Symbols/PlaceholderLocalSymbol.cs

```diff
@@ -88,11 +88,43 @@ internal override EELocalSymbolBase ToOtherMethod(MethodSymbol method, TypeMap t
 
         internal static BoundExpression ConvertToLocalType(CSharpCompilation compilation, BoundExpression expr, TypeSymbol type, DiagnosticBag diagnostics)
         {
+            if (type.IsPointerType())
+            {
+                var syntax = expr.Syntax;
+                var intPtrType = compilation.GetSpecialType(SpecialType.System_IntPtr);
+                Binder.ReportUseSiteDiagnostics(intPtrType, diagnostics, syntax);
+                MethodSymbol conversionMethod;
+                if (Binder.TryGetSpecialTypeMember(compilation, SpecialMember.System_IntPtr__op_Explicit_ToPointer, syntax, diagnostics, out conversionMethod))
+                {
+                    var temp = ConvertToLocalTypeHelper(compilation, expr, intPtrType, diagnostics);
```

Comment:
Why it should be a boxed `IntPtr`. For example, reflection API use `System.Reflection.Pointer` to represent pointers boxed in `object` type (e.g. to represent a return value or a parameter of a pointer type in a method invocation). Why use different convention here?

What is currently returned by the debugger?